src/backend/cuda/kernel/index.cuh

```diff
@@ -43,13 +43,17 @@ __global__ void index(Param<T> out, CParam<T> in, const IndexKernelParam p,
         gw < out.dims[3]) {
         // calculate pointer offsets for input
         int i =
-            p.strds[0] * trimIndex(s0 ? gx + p.offs[0] : ptr0[gx], in.dims[0]);
+            p.strds[0] *
+            trimIndex(s0 ? gx * p.steps[0] + p.offs[0] : ptr0[gx], in.dims[0]);
```

Comment:
Do you need to update the condition on 42? looks like you would read out of bounds here.

Reply:
The condition is still correct because `gx gy gz gw` refer to indices in the output array, and the given check makes sure they are always inside the output array dimensions. The changes affect the indexing of the input array being sampled and the function `trimIndex` makes sure it's inside the input array dimensions.